api/v1/_submissions/PKPBackendSubmissionsHandler.inc.php

```diff
@@ -67,109 +76,129 @@ public function authorize($request, &$args, $roleAssignments)
 
-        // Merge query params over default params
-        $defaultParams = [
-            'count' => 20,
-            'offset' => 0,
-        ];
+        if (!$context) {
+            return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
+        }
+
+        $collector = $this->getSubmissionCollector($slimRequest->getQueryParams());
+
+        HookRegistry::call('API::submissions::params', [$collector, $slimRequest]);
 
-        // Anyone not a manager or site admin can only access their assigned
-        // submissions
-        $userRoles = $this->getAuthorizedContextObject(ASSOC_TYPE_USER_ROLES);
+        // Prevent users from viewing submissions they're not assigned to,
+        // except for journal managers and admins.
+        $userRoles = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_USER_ROLES);
         $canAccessUnassignedSubmission = !empty(array_intersect([ROLE_ID_SITE_ADMIN, ROLE_ID_MANAGER], $userRoles));
-        if (!$canAccessUnassignedSubmission) {
-            $defaultParams['assignedTo'] = [$currentUser->getId()];
+        if (!$canAccessUnassignedSubmission && !in_array($currentUser->getId(), $collector->assignedTo)) {
+            return $response->withStatus(403)->withJsonError('api.submissions.403.requestedOthersUnpublishedSubmissions');
         }
 
-        $params = array_merge($defaultParams, $slimRequest->getQueryParams());
+        $submissions = Query::submission()->getMany($collector);
+
+        $userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /** @var UserGroupDAO $userGroupDao */
+        $userGroups = $userGroupDao->getByContextId($context->getId())->toArray();
+
+        return $response->withJson([
+            'itemsMax' => Query::submission()->getCount($collector),
+            'items' => Map::submissionToSchema()->mapManyToSubmissionsList($submissions, $userGroups),
+        ], 200);
+    }
+
+    /**
+     * Configure a submission Collector based on the query params
+     */
+    protected function getSubmissionCollector(array $queryParams): Collector
+    {
+        $request = Application::get()->getRequest();
+        $currentUser = $request->getUser();
+        $context = $request->getContext();
 
-        // Process query params to format incoming data as needed
-        foreach ($params as $param => $val) {
+        $collector = Query::submission()->getCollector()
+            ->filterByContextIds([$context->getId()])
+            ->limit(30)
+            ->offset(0);
+
+        // Only admins and managers may access submissions they are not assigned to
+        $userRoles = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_USER_ROLES);
+        $canAccessUnassignedSubmission = !empty(array_intersect([ROLE_ID_SITE_ADMIN, ROLE_ID_MANAGER], $userRoles));
+        if (!$canAccessUnassignedSubmission) {
+            $collector->assignedTo([$currentUser->getId()]);
+        }
+
+        foreach ($queryParams as $param => $val) {
```

Comment:
We could use an approach here to use [`when`](https://laraveldaily.com/less-know-way-conditional-queries/) clauses.

Something like this:
```
foreach ($queryParams as $param => $val) {
    $collector->when($param === 'orderby', function ($q) use ($val, $queryParams, $collector) {
        $availableValues = [
            $collector::ORDERBY_DATE_PUBLISHED => 1,
            $collector::ORDERBY_DATE_SUBMITTED => 1,
            $collector::ORDERBY_LAST_ACTIVITY => 1,
            $collector::ORDERBY_LAST_MODIFIED => 1,
            $collector::ORDERBY_SEQUENCE => 1,
            $collector::ORDERBY_TITLE => 1,
        ];
        if (!isset($availableValues[$val])) {
            return $q;
        }

        $direction = $collector::ORDER_DIR_DESC;

        if (isset($queryParams['orderDirection']) && $queryParams['orderDirection'] === $collector::ORDER_DIR_ASC) {
            $direction = $collector::ORDER_DIR_ASC;
        }

        return $q->orderBy($val, $direction);
    });
    $collector->when($param === 'status', function ($q) use ($val) {
        return $q->filterByStatus(array_map('intval', $this->paramToArray($val)));
    });
    $collector->when($param === 'stageIds', function ($q) use ($val) {
        return $q->filterByStageIds(array_map('intval', $this->paramToArray($val)));
    });
    $collector->when($param === 'assignedTo', function ($q) use ($val) {
        $val = array_map('intval', $this->paramToArray($val));
    
        if ($val === [-1]) {
            $val = array_shift($val);
        }
    
        return $q->assignedTo($val);
    });
    $collector->when($param === 'daysInactive', function ($q) use ($val) {
        return $q->filterByDaysInactive((int) $val);
    });
    $collector->when($param === 'offset', function ($q) use ($val) {
        return $q->offset((int) $val);
    });
    $collector->when($param === 'searchPhrase', function ($q) use ($val) {
        return $q->searchPhrase($val);
    });
    $collector->when($param === 'count', function ($q) use ($val) {
        return $q->limit(min(self::MAX_COUNT, (int) $val));
    });
    $collector->when($param === 'isIncomplete', function ($q) {
        return $q->filterByIncomplete(true);
    });
    $collector->when($param === 'isOverdue', function ($q) {
        return $q->filterByOverdue(true);
    });
}
```

Reply:
This would couple the `Collector` with the the `Handler`. One of our goals is to decouple the code from the handlers, so we can reuse it anywhere (eg - CLI, plugins, etc.)